src/common/snippets/src/lowered/expression.cpp

```diff
@@ -130,30 +133,40 @@ void Expression::set_loop_ids(const std::vector<size_t>& loops) {
     m_loop_ids = loops;
 }
 
-void Expression::update_node_and_connectors(const std::vector<PortConnectorPtr>& new_inputs,
-                                            const std::shared_ptr<Node>& new_node) {
-    OPENVINO_ASSERT(m_source_node->get_type_info() == new_node->get_type_info(),
-                    "Can't clone expression for a new node with incompatible type");
+void Expression::update_port_attributes(const std::shared_ptr<Node>& new_node, const std::vector<PortConnectorPtr>& new_inputs,
+                                        const std::vector<PortDescriptorPtr>& new_in_descs, const std::vector<PortDescriptorPtr>& new_out_descs) {
+    OPENVINO_ASSERT(m_source_node->get_type_info() == new_node->get_type_info(), "Can't clone expression for a new node with incompatible type");
     m_source_node = new_node;
-    OPENVINO_ASSERT(new_inputs.size() == m_input_port_descriptors.size(),
-                    "Can't create Expression with new inputs: invalid number of input port connectors passed");
+    OPENVINO_ASSERT(new_inputs.size() == new_in_descs.size(), "Can't create Expression with new inputs: invalid number of input port connectors passed");
+    m_input_port_descriptors = new_in_descs;
     m_input_port_connectors = new_inputs;
     for (size_t i = 0; i < m_input_port_descriptors.size(); i++) {
         const auto& i_con = new_inputs[i];
         const auto& i_port = get_input_port(i);
         if (!i_con->found_consumer(i_port))
             i_con->add_consumer(i_port);
     }
+    m_output_port_descriptors = new_out_descs;
     m_output_port_connectors.resize(m_output_port_descriptors.size());
     for (size_t i = 0; i < m_output_port_descriptors.size(); i++) {
         m_output_port_connectors[i] = std::make_shared<PortConnector>(get_output_port(i));
     }
 }
 
-ExpressionPtr Expression::clone_with_new_inputs(const std::vector<PortConnectorPtr>& new_inputs,
-                                                const std::shared_ptr<Node>& new_node) const {
-    const auto& expr = std::shared_ptr<Expression>(new Expression(*this));
-    expr->update_node_and_connectors(new_inputs, new_node);
+ExpressionPtr Expression::clone_with_new_inputs(const std::shared_ptr<Node>& new_node,
+                                                const std::vector<PortConnectorPtr>& new_inputs,
+                                                const std::vector<PortDescriptorPtr>& new_in_descs) const {
+    auto clone_ports_descriptors = [](const std::vector<PortDescriptorPtr>& src) {
+        std::vector<PortDescriptorPtr> dst(src.size());
+        for (size_t i = 0; i < src.size(); i++)
+            dst[i] = src[i]->clone();
+        return dst;
+    };
+    const auto& expr = clone();
+    const auto& in_descs = !new_in_descs.empty() ? new_in_descs : clone_ports_descriptors(m_input_port_descriptors);
```

Comment:
Minor: can we describe in the method's description what the method does in case of empty `new_in_descs`?